phoenix-tools/src/main/java/org/apache/phoenix/schema/SchemaExtractionProcessor.java

```diff
@@ -374,20 +397,49 @@ private String getColumnInfoStringForView(PTable table, PTable baseTable) {
 
     private String extractColumn(PColumn column) {
         String colName = column.getName().getString();
+        if (column.getFamilyName() != null){
+            String colFamilyName = column.getFamilyName().getString();
+            // check if it is default column family name
+            colName = colFamilyName.equals(QueryConstants.DEFAULT_COLUMN_FAMILY)? colName : String.format("\"%s\".\"%s\"", colFamilyName, colName);
+        }
+        boolean isArrayType = column.getDataType().isArrayType();
         String type = column.getDataType().getSqlTypeName();
+        Integer maxLength = column.getMaxLength();
+        Integer arrSize = column.getArraySize();
+        Integer scale = column.getScale();
         StringBuilder buf = new StringBuilder(colName);
         buf.append(' ');
-        buf.append(type);
-        Integer maxLength = column.getMaxLength();
-        if (maxLength != null) {
-            buf.append('(');
-            buf.append(maxLength);
-            Integer scale = column.getScale();
-            if (scale != null) {
-                buf.append(',');
-                buf.append(scale); // has both max length and scale. For ex- decimal(10,2)
+
+        if (isArrayType) {
+            String arrayPrefix = type.split("\\s+")[0];
+            buf.append(arrayPrefix);
+            if (maxLength != null) {
```

Comment:
nit: this if condition is duplicated in the else part. Please refactor it by extracting common parts and executing specific Array related logic when if clause passes.